src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetUptime.cs

```diff
@@ -0,0 +1,82 @@
+/********************************************************************++
+Copyright (c) Microsoft Corporation.  All rights reserved.
+--********************************************************************/
+
+using System;
+using System.Management.Automation;
+using System.Diagnostics;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// This class implements Get-Uptime.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "Uptime", DefaultParameterSetName = "Timespan", HelpUri = "")]
+    public class GetUptimeCommand : PSCmdlet
+    {
+        /// <summary>
+        /// Timespan parameter
+        /// Time since the system started up
+        /// </summary>
+        /// <value></value>
+        [Parameter(ParameterSetName = "Timespan")]
+        public SwitchParameter Timespan { get; set; } = new SwitchParameter();
+
+        /// <summary>
+        /// Since parameter
+        /// The system startup time
+        /// </summary>
+        /// <value></value>
+        [Parameter(ParameterSetName = "Since")]
+        public SwitchParameter Since { get; set; } = new SwitchParameter();
+
+        /// <summary>
+        /// Pretty parameter
+        /// Convert output to default formated string
+        /// </summary>
+        /// <value></value>
+        [Parameter(ParameterSetName = "Timespan")]
+        [Parameter(ParameterSetName = "Since")]
+        public SwitchParameter Pretty { get; set; } = new SwitchParameter();
+
+        /// <summary>
+        /// ProcessRecord() override.
+        /// This is the main entry point for the cmdlet.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            // Get-Uptime return null if IsHighResolution = false  
+            // because stopwatch.GetTimestamp() return DateTime.UtcNow.Ticks
+            // instead of ticks from system startup
```

Comment:
"silently" is from @daxian-dbw in #2507. I agree that user must get a error message about a problem. But in this case it is a question because such event is likely _very rare_ because all modern platforms should work correctly. On the other hand, the user can always change behavior using `-ErrorAction`  @daxian-dbw please comment too.

Windows has only one correct way to get uptime that is 'System Uptime' performance counter. But performance counters still are not ported. I could not find anything other.